lstchain/calib/camera/time_sampling_correction.py

```diff
@@ -0,0 +1,118 @@
+import h5py
+import numpy as np
+
+
+from ctapipe.core import Component
+from ctapipe.core.traits import Unicode
+
+__all__ = [
+    'TimeSamplingCorrection',
+    ]
+
+
+
+class TimeSamplingCorrection(Component):
+    """
+        The PulseTimeCorrection class to correct time pulse
+        using Fourier series expansion.
+    """
+
+    time_sampling_correction_path = Unicode(
+        '',
+        help='Path to the waveform sampling correction file',
+        allow_none = True,
+    ).tag(config=True)
+
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
+        self.time_sampling_coefficients = None
+
+        self.load_sampling_coefficient_file()
+
+    def load_sampling_coefficient_file(self):
+        """
+            Function to load sampling coefficient file.
+        """
+
+        try:
+            with h5py.File(self.time_sampling_correction_path, 'r') as hf:
+                self.time_sampling_coefficients = np.array(hf['sampling_interval_coefficient'])
+        except:
+            self.log.error(f"Problem in reading sampling coefficient file {self.time_sampling_correction_path}")
+
+
+    def get_corrections(self, event, telid):
+        """
+        Get the time/charge sampling corrections for one event and one telescope
+
+        Parameters
+        ----------
+        event: general event container
+        telid: id of the telescope
+
+        Return
+        ------
+        sampling_corrections: np.array (n_gains, n_pixels, n_samples) with the correction factors
+        """
+
+        n_gains = 2
+        n_modules = 265  # number of modules in LST's camera.
+        n_pixel_per_module = 7  # number of pixels per one module.
+        size_drs4 = 1024
+        n_pixels = 1855
+        roi = 36
+
+        # shift the first capacitor to the value used in the r1
+        r1_roi_start = 3
+
+        fc_all = event.lst.tel[telid].evt.first_capacitor_id
+
+        # get the first capacitor per pixel and gain
```

Comment:
This should be its own function. Also, isn't this already implemented in the `Calibrator`? If that's the same, define a function and use this function in both places.

Reply:
Same comment as above. I plan to update the code using the new reader functions as soon as we move to it